llvm/test/CodeGen/ARM/arm-shrink-wrapping.ll

```diff
@@ -1965,32 +1965,34 @@ define float @debug_info(float %gamma, float %slopeLimit, i1 %or.cond, double %t
 ; ARM-ENABLE-NEXT:  @ %bb.1: @ %bb3
 ; ARM-ENABLE-NEXT:    push {r4, r7, lr}
 ; ARM-ENABLE-NEXT:    add r7, sp, #4
-; ARM-ENABLE-NEXT:    sub r4, sp, #16
+; ARM-ENABLE-NEXT:    sub r4, sp, #24
```

Comment:
Any idea what's up with this increased size of stack frame? I can see that it goes with a newly introduced spill and reload of d10 to the `[r4, #16]` slot, but why has this change created the need for an extra spill at all? Has pre-RA scheduling somehow increased pressure?

Reply:
I *think* that the reason is that DAG scheduler placed VADD before the call to `pow` and this increased register pressure over the limit. (Unfortunately, *machine* scheduler can't reorder instructions around calls.)